common/Services/NotificationService.cs

```diff
@@ -0,0 +1,240 @@
+                {
+                    // Launch compmgmt.msc in powershell
+                    var psi = new ProcessStartInfo();
+                    psi.FileName = "powershell";
+                    psi.Arguments = "Start-Process compmgmt.msc -Verb RunAs";
+                    Process.Start(psi);
+                }
+            }
+            catch (Exception ex)
+            {
+                _log.Error(_componentName, $"Unable to launch computer management due to exception", ex);
+            }
+        }
+    }
+
+    public void ShowRestartNotification()
+    {
+        if (_notificationQueue != null)
+        {
+            var command = new RelayCommand(() =>
+            {
+                var startInfo = new System.Diagnostics.ProcessStartInfo();
+                startInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+
+                // Restart the computer
+                // To Do: Remove the hardcoded time
+                startInfo.FileName = "shutdown.exe";
+                startInfo.Arguments = "-r -t 0";
+                startInfo.Verb = string.Empty;
+
+                var process = new System.Diagnostics.Process();
+                process.StartInfo = startInfo;
+                process.Start();
+            });
+
+            _dispatcher.EnqueueAsync(() =>
+                _notificationQueue?.Show(new Notification
+                {
+                    Title = _stringResource.GetLocalized("HyperVErrorTitle", _microsoftText, _hyperVText),
+                    Message = _stringResource.GetLocalized("RestartMessage", _hyperVText),
+                    Severity = InfoBarSeverity.Warning,
+                    ActionButton = new Button
+                    {
+                        Content = _stringResource.GetLocalized("RestartButton"),
+                        Command = command,
+                    },
+                }));
+        }
+        else
+        {
+            _log.Error(_componentName, "Notification queue is not initialized");
+        }
+    }
+
+    private void ShowUnableToAddToHyperVAdminGroupNotification()
+    {
+        ShowNotificationAsync(
+            _stringResource.GetLocalized("HyperVErrorTitle", _microsoftText, _hyperVText),
+            _stringResource.GetLocalized("UserAddHyperVAdminFailed", _hyperVText),
+            InfoBarSeverity.Warning).Wait();
+    }
+
+    public void CheckIfUserIsAHyperVAdminAndShowNotification()
+    {
+        if (!_windowsIdentityService.IsUserHyperVAdmin())
+        {
+            if (_notificationQueue != null)
+            {
+                var notification = new Notification();
+
+                var command = new RelayCommand(() =>
+                {
+                    var startInfo = new System.Diagnostics.ProcessStartInfo();
+                    startInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+                    startInfo.FileName = "net.exe";
```

Comment:
I think we should construct full path to the executable (here and above) to avoid calling some executable with the same name if executable search finds it in some other location (if PATH was modified for example).

Reply:
Wouldn't system32 still be on top for those PATHs? 
Either way, changed.